core/data/src/main/java/com/nolawiworkineh/data/networking/TokenAuthenticator.kt

```diff
@@ -0,0 +1,53 @@
+package com.nolawiworkineh.data.networking
+
+import com.nolawiworkineh.core.data.networking.AccessTokenRequest
+import com.nolawiworkineh.core.domain.SessionStorage
+import kotlinx.coroutines.runBlocking
+import okhttp3.Authenticator
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.Route
+import timber.log.Timber
+import javax.inject.Inject
+
+class TokenAuthenticator @Inject constructor(
+    private val sessionStorage: SessionStorage,
+    private val tokenApi: TokenApi
+) : Authenticator {
+    override fun authenticate(route: Route?, response: Response): Request? {
+        // Don't retry if we failed to refresh the token
+        if (response.request.url.encodedPath.contains("accessToken")) {
+            return null
+        }
```

Comment:
Isn't this already made sure by not adding this authenticator to the token endpoints?